internal/controllers/configuration/secret_controller.go

```diff
@@ -0,0 +1,127 @@
+package configuration
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/dataplane"
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/util"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+// -----------------------------------------------------------------------------
+// CoreV1 Secret - Reconciler
+// -----------------------------------------------------------------------------
+
+// CoreV1SecretReconciler reconciles Secret resources
+type CoreV1SecretReconciler struct {
+	client.Client
+
+	Log              logr.Logger
+	Scheme           *runtime.Scheme
+	DataplaneClient  *dataplane.KongClient
+	CacheSyncTimeout time.Duration
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *CoreV1SecretReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	c, err := controller.New("CoreV1Secret", mgr, controller.Options{
+		Reconciler: r,
+		LogConstructor: func(_ *reconcile.Request) logr.Logger {
+			return r.Log
+		},
+		CacheSyncTimeout: r.CacheSyncTimeout,
+	})
+	if err != nil {
+		return err
+	}
+
+	predicateFuncs := predicate.NewPredicateFuncs(r.shouldReconcileSecret)
+	//we should always try to delete secrets in caches when they are deleted in cluster.
+	predicateFuncs.DeleteFunc = func(event event.DeleteEvent) bool { return true }
+	return c.Watch(
+		&source.Kind{Type: &corev1.Secret{}},
+		&handler.EnqueueRequestForObject{},
+		predicateFuncs,
+	)
+}
+
+// shouldReconcileSecret is the filter function to judge whether the secret should be reconciled
+// and stored in cache of the controller. It returns true for the secret should be reconciled when:
+// - the secret has label: konghq.com/ca-cert:true
+// - or the secret is referred by objects we care (service, ingress, gateway, ...)
+
+func (r *CoreV1SecretReconciler) shouldReconcileSecret(obj client.Object) bool {
+	secret, ok := obj.(*corev1.Secret)
+	if !ok {
+		return false
+	}
+
+	labels := secret.Labels
+	if labels != nil && labels["konghq.com/ca-cert"] == "true" {
```

Comment:
There's a constant for the label in `internal/store/store.go` (`caCertKey`). Do you mind making it public and using it here?

Reply:
IMO importing the `store` package just to use the value but not to use the object stores inside the packet is a bit weird, and will mislead readers of the code to think that the secret controller uses functions in store directly. Instead, I defined the constant in the file.